api/response/response.go

```diff
@@ -27,6 +31,14 @@ func Write(w http.ResponseWriter, resp Response) {
 	return
 }
 
+func WriteErr(ctx context.Context, w http.ResponseWriter, resp Response) {
```

Comment:
every call to WriteErr is passing `writeErr(ctx.Req.Context(), ctx, error)`
so why not just use 
```
func WriteErr(c *macaron.Context, resp Response) {
  ctx := c.Req.Context()
  .....
}
```
It may even be a better idea to just include the error logger in the Write() method, eg.
```
func Write(ctx *macaron.Context, resp Response) {
   defer resp.Close()
	body, err := resp.Body()
	if err != nil {
		w.WriteHeader(http.StatusInternalServerError)
		w.Write([]byte(err.Error()))
		return
	}
	for k, v := range resp.Headers() {
		w.Header().Set(k, v)
	}
	w.WriteHeader(resp.Code())
	w.Write(body)
        
        // log errors
        if errResp, ok := resp.(Error); ok { // or maybe "if resp.Code() >= 400"
            span := opentracing.SpanFromContext(ctx)
 	    body, _ := resp.Body()
 	    span.LogFields(log.String("error.kind", string(body)))
 	     tags.Error.Set(span, true)
    }
}
```